File: go/leap/leap.go
package leap

const TestVersion = 1

func IsLeapYear(year int) bool {
	yearDivisbleBy := func(x int) bool {
		return (year % x) == 0
	}

	return yearDivisbleBy(4) && (!yearDivisbleBy(100) || yearDivisbleBy(400))
}

File: ruby/leap/leap.rb
class Fixnum
  def devisible_by?(num)
    self % num == 0
  end
end

class Year
  VERSION = 1
  def self.leap?(y)
    y.devisible_by?(400) || (!y.devisible_by?(100) && y.devisible_by?(4))
  end
end
